### PR TITLE
[Mosip-32572] java21 migration

### PR DESCRIPTION
**Pull request by Khuddusshariff0022**

### CODE REVIEW

File: data-share/data-share-service/pom.xml [hunk 1]
```diff
@@ -1,255 +1,240 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-configuration-processor</artifactId>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>3.11.2</version>
```
Comment (sowmya695) [inline data-share/data-share-service/pom.xml:140]: Please add the version above in properties

File: data-share/data-share-service/src/test/java/io/mosip/datashare/test/config/TestSecurityConfig.java [hunk 2]
```diff
@@ -4,15 +4,15 @@
...
 
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
-import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
 import org.springframework.security.config.annotation.web.builders.WebSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
-import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+//import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
```
Comment (sowmya695) [inline data-share/data-share-service/src/test/java/io/mosip/datashare/test/config/TestSecurityConfig.java:15]: Remove the unused import

File: data-share/data-share-service/src/test/java/io/mosip/datashare/test/config/TestSecurityConfig.java [hunk 3]
```diff
@@ -25,34 +25,34 @@
...
 	}
 
-	@Override
-	protected void configure(final HttpSecurity httpSecurity) throws Exception {
-		httpSecurity.csrf().disable();
-		httpSecurity.httpBasic().and().authorizeRequests().anyRequest().authenticated().and().sessionManagement()
-				.sessionCreationPolicy(SessionCreationPolicy.STATELESS).and().exceptionHandling()
-				.authenticationEntryPoint(unauthorizedEntryPoint());
-	}
+//	@Override
```
Comment (sowmya695) [inline data-share/data-share-service/src/test/java/io/mosip/datashare/test/config/TestSecurityConfig.java:49]: Remove the commented code

File: data-share/data-share-service/src/test/java/io/mosip/datashare/test/controller/DataShareControllerTest.java [hunk 4]
```diff
@@ -45,94 +46,95 @@
...
+        multiPartFile = new MockMultipartFile("file", "NameOfTheFile", "multipart/form-data",
+                new ByteArrayInputStream(dataBytes));
+
+    }
+
+    @Test
+    @WithUserDetails("test")
+    public void testDataShareSuccess() throws Exception {
+        DataShare dataShare=new DataShare();
+//		Mockito.when(
```
Comment (sowmya695) [inline data-share/data-share-service/src/test/java/io/mosip/datashare/test/controller/DataShareControllerTest.java:90]: Removed the commented code

File: data-share/data-share-service/pom.xml (hunk 1 above)
Comment (Khuddusshariff0022) [reply data-share/data-share-service/pom.xml:140]: done

File: data-share/data-share-service/src/test/java/io/mosip/datashare/test/config/TestSecurityConfig.java (hunk 2 above)
Comment (Khuddusshariff0022) [reply data-share/data-share-service/src/test/java/io/mosip/datashare/test/config/TestSecurityConfig.java:15]: done

File: data-share/data-share-service/src/test/java/io/mosip/datashare/test/config/TestSecurityConfig.java (hunk 3 above)
Comment (Khuddusshariff0022) [reply data-share/data-share-service/src/test/java/io/mosip/datashare/test/config/TestSecurityConfig.java:49]: done

**Review by LoganathanSekar7627 — APPROVED** [review]

**Pull request merged**